app/com/linkedin/drelephant/tuning/AutoTuningAPIHelper.java

```diff
@@ -55,6 +58,14 @@
       "autotuning.default.allowed_max_resource_usage_percent";
   private static final String ALLOWED_MAX_EXECUTION_TIME_PERCENT_DEFAULT =
       "autotuning.default.allowed_max_execution_time_percent";
+  private static final String TUNEIN_RE_ENABLE_ALLOWED =
```

Comment:
I think re enablement of job should be separate thread like (FitnessCompute) . Getcurrentrunparameter call should not have logic for re- neablement. There are multiple design issue in this 
1) getCurrentRunParameter should be light weight call to get the parameters , ideally it should not be doing anything. 
2) There should be separate responsibility of Tuning Threads and getCurrentRunParameter. 

Ideally a thread should run daily(or on some frequency)  and re enable the eligible jobs . This way getCurrentRunParameter will be having only task of getting parameters , nothing else. 

Also there should be several metadata which should be added to job properties(tuningjobdefinitation) table
1) Number of time re enablement occurred for this job
2) Last time when the job was re enabled